test/recipes/70-test_key_share.t

```diff
@@ -188,6 +188,26 @@ $testtype = TRAILING_DATA;
 $proxy->start();
 ok(TLSProxy::Message->fail(), "key_share trailing data in ServerHello");
 
+#Test 18: key_share should not be sent if the client is not capable of
+#         negotiating TLSv1.3
+$proxy->clear();
+$proxy->filter(undef);
+$proxy->clientflags("-no_tls1_3");
+$proxy->start();
+my $clienthello = ${$proxy->message_list}[0];
```

Comment:
Tip of the day, this works as well:

```
my $clienthello = $proxy->message_list->[0];
```

I tend to prefer a style with less {} when possible, but YMMV.